content/en/blog/_posts/2024-02-22-diy-create-your-own-cloud-with-kubernetes-part-1/index.md

```diff
@@ -0,0 +1,158 @@
+---
+layout: blog
+title: "DIY: Create Your Own Cloud with Kubernetes (Part 1)"
+slug: diy-create-your-own-cloud-with-kubernetes-part-1
+date: 2024-01-22
+---
+
+**Author**: Andrei Kvapil (Ænix)
+
+*We have a deep affection for Kubernetes and dream that all modern technologies will soon start utilizing its remarkable patterns.*
+
+Have you ever thought about building your own cloud? I bet you have. But is it possible to do this using only modern technologies and approaches, without leaving the cozy Kubernetes ecosystem? Our experience in developing Cozystack required us to delve deeply into it.
+
+You might argue that Kubernetes is not intended for this purpose and why not simply use OpenStack for bare metal servers and run Kubernetes inside it as intended. But by doing so, you would simply shift the responsibility from your hands to the hands of OpenStack administrators. This would add at least one more huge and complex system to your ecosystem.
+
+Why complicate things? - after all, Kubernetes already has everything needed to run tenant Kubernetes clusters at this point.
+
+I want to share with you our experience in developing a cloud platform based on Kubernetes, highlighting the open-source projects that we use ourselves and believe deserve your attention.
+
+In this series of articles, I will tell you our story about how we prepare managed Kubernetes from bare metal using only open-source technologies. Starting from the basic level of data center preparation, running virtual machines, isolating networks, setting up fault-tolerant storage to provisioning full-featured Kubernetes clusters with dynamic volume provisioning, load balancers, and autoscaling.
+
+With this article, I start a series consisting of several parts:
+
+- **Part 1**: Preparing the groundwork for your cloud. Challenges faced during the preparation and operation of Kubernetes on bare metal and a ready-made recipe for provisioning infrastructure.
+- **Part 2**: Networking, storage, and virtualization. How to turn Kubernetes into a tool for launching virtual machines and what is needed for this.
+- **Part 3**: Cluster API and how to start provisioning Kubernetes clusters at the push of a button. How autoscaling works, dynamic provisioning of volumes, and load balancers.
```

Comment:
Let's get all three parts lined up with publication dates assigned and ready to merge, before we publish the first in the series.

Otherwise, we're implying a promise we might not keep.